### PR TITLE
Feature: Adjust display for share grant events

### PR DESCRIPTION
**Pull request by greg-adams**
## Description
- Update display for share activity events
  -  Migrate to common activity item for grant followers, notes, and share events (`UserActivityItem`)
- Align styling for sidebar components

## Screenshots / Demo Video
Screenshot from 2024-10-02 10-33-27

### Automated and Unit Tests
- [x] Added Unit tests

### Manual tests for Reviewer
- [ ] Added steps to test feature/functionality manually

## Checklist
- [x] Provided ticket and description
- [x] Provided screenshots/demo
- [ ] Provided testing information
- [x] Provided adequate test coverage for all new code
- [x] Added PR reviewers

### CODE REVIEW

File: packages/client/src/components/GrantNotes.vue [hunk 1]
```diff
@@ -50,13 +50,19 @@
...
+    <UserActivityItem
       v-if="userNote && !editingNote"
-      data-test-user-note
       :class="userNoteClass"
-      :note="userNote"
+      :user-name="userNote.user.name"
+      :user-email="userNote.user.email"
+      :team-name="userNote.user.team.name"
+      :avatar-color="userNote.user.avatarColor"
+      :created-at="userNote.createdAt"
```
Comment (TylerHendrickson) [inline packages/client/src/components/GrantNotes.vue:61]: This block appears to be missing something like `:is-edited="userNote.isRevised"` – when testing in my local environment, I'm missing the `(edited)` part of the `{{ timeElapsedString }} (edited)` part of the [template](https://github.com/usdigitalresponse/usdr-gost/blob/20dc496537347859d82798c9e65eacfb7e990db5/packages/client/src/components/UserActivityItem.vue#L36-L42).

File: packages/client/src/components/GrantNotes.vue [hunk 2]
```diff
@@ -70,18 +76,26 @@
...
-        />
+        <UserActivityItem
+          class="activity-container"
+          :user-name="note.user.name"
+          :user-email="note.user.email"
+          :team-name="note.user.team.name"
+          :avatar-color="note.user.avatarColor"
+          :created-at="note.createdAt"
+          copy-email-enabled
+          :data-test-other-note-id="note.id"
```
Comment (TylerHendrickson) [inline packages/client/src/components/GrantNotes.vue:95]: Also missing an `:is-edited` prop (see previous comment)

**Review by TylerHendrickson — CHANGES_REQUESTED** [review]
@greg-adams Things are looking really nice in my local environment! 

I only found one issue during testing (see my comments). Additionally, I'm wondering  the [`packages/client/src/components/GrantNote.vue` component](https://github.com/usdigitalresponse/usdr-gost/blob/20dc496537347859d82798c9e65eacfb7e990db5/packages/client/src/components/GrantNote.vue) is still being used anywhere. If not, can we delete that file (along with its tests) in this PR as well?

File: packages/client/src/components/GrantNotes.vue (hunk 1 above)
Comment (greg-adams) [reply packages/client/src/components/GrantNotes.vue:61]: thanks for catching, I added this back in

**Review by TylerHendrickson — APPROVED** [review]

**Comment by TylerHendrickson** [comment]
@greg-adams Looks good!

**Pull request merged**